src/quo2/components/bottom_sheet/view.cljs

```diff
@@ -0,0 +1,174 @@
+                             backdrop-dismiss?  true
+                             expandable?        true}} props
+        content-height (reagent/atom nil)
+        visible-internal? (reagent/atom visible?)
+        show-bottom-sheet? (reagent/atom false)
+        keyboard-was-shown (reagent/atom false)
+        expanded? (reagent/atom nil)]
+    [safe-area/consumer
+     (fn [insets]
+       [:f>
+        (fn []
+          (let [{window-height :height
+                 window-width :width} (rn/use-window-dimensions)
+                {:keys [keyboard-shown]} (hooks/use-keyboard)
+                bg-height-expanded (- window-height (:top insets))
+                bg-height (max (min @content-height bg-height-expanded) 200)
+                bottom-sheet-dy (reanimated/use-shared-value 0)
+                pan-y (reanimated/use-shared-value 0)
+                translate-y (.useTranslateY ^js bottom-sheet-js window-height bottom-sheet-dy pan-y)
+                bg-opacity (.useBackgroundOpacity ^js bottom-sheet-js translate-y bg-height window-height)
+                on-content-layout   (fn [evt]
+                                      (let [height  (oget evt "nativeEvent" "layout" "height")]
+                                        (reset! content-height height)))
+                close-bottom-sheet (fn []
+                                     (reset! visible-internal? false)
+                                     (when (fn? on-cancel) (on-cancel))
+                                     (timer/set-timeout
+                                      #(do (re-frame/dispatch [:bottom-sheet/hide]))
+                                      animation-delay))
+                on-expanded (fn []
+                              (reanimated/set-shared-value bottom-sheet-dy bg-height-expanded)
+                              (reanimated/set-shared-value pan-y 0))
+                on-collapsed (fn []
+                               (reanimated/set-shared-value bottom-sheet-dy bg-height)
+                               (reanimated/set-shared-value pan-y 0))
+                bottom-sheet-gesture (get-bottom-sheet-gesture pan-y translate-y bg-height bg-height-expanded window-height keyboard-shown disable-drag? expandable? show-bottom-sheet? expanded? close-bottom-sheet)]
+
+            (react/effect! #(do
+                              (cond
+                                (and
+                                 (not @show-bottom-sheet?)
+                                 @visible-internal?
+                                 (some? @content-height)
+                                 (> @content-height 0))
+                                (reset! show-bottom-sheet? true)
+
+                                (and @show-bottom-sheet? (not @visible-internal?))
+                                (reset! show-bottom-sheet? false)))
+                           [@show-bottom-sheet? @visible-internal? @content-height])
+            (react/effect! #(do
```

Comment:
i gues you didn't move effect! to react-native namespace because of macros ?

Reply:
yes, i'm not that familiar with clojure macros, low-level stuff. Would be awesome, if somebody could move the react functions to quo2 namespace

Reply:
will do it , that's fine to merge this PR for now